src/pages/goal-details/goal-details.ts

```diff
@@ -30,4 +32,22 @@ ionViewDidLoad() {
   editSettingsTapped(event) {
     this.navCtrl.push(EditGoalPage, this.goal);
   }
+
+  //datapoints have a timestamp, daystamp, value, comment, and requestid
+  addDataPoint(goal){
+    //create timestamp for goal
+    let decade = 60 * 60 * 24 * 365 * 10;
+		let d = new Date();
+		let t = Math.floor(d.getTime() / 1000);
+
+		var goaldate = t + decade;
+
+    let datapoint = {
+      timestamp: goaldate,
+      value: this.datapoint.value,
+      comment: this.datapoint.comment,
```

Comment:
This might cause an error if the comment is undefined. The timestamp is not a the goal date. I think you can leave it out.